packages/gestalt/src/Spinner/VRSpinner.tsx

```diff
@@ -6,40 +7,88 @@ import { useDefaultLabelContext } from '../contexts/DefaultLabelProvider';
 const SIZE_NAME_TO_PIXEL = {
   sm: 32,
   md: 40,
-  lg: 48,
+  lg: 56,
 } as const;
 
-type Props = {
-  accessibilityLabel?: string;
-  delay?: boolean;
+type SpinnerBodyProps = {
+  accessibilityLabel: string;
+  delay: boolean;
   show: boolean;
-  size?: 'sm' | 'md' | 'lg';
+  size: 'sm' | 'md' | 'lg';
+  onExitAnimationEnd: () => void;
 };
 
-export default function Spinner({ accessibilityLabel, delay = true, show, size = 'md' }: Props) {
-  const { accessibilityLabel: accessibilityLabelDefault } = useDefaultLabelContext('Spinner');
-
-  if (!show) return null;
-
+function SpinnerBody({
+  accessibilityLabel,
+  delay,
+  show,
+  size,
+  onExitAnimationEnd,
+}: SpinnerBodyProps) {
   return (
     <Box display="flex" justifyContent="around">
       <div
-        aria-label={accessibilityLabel ?? accessibilityLabelDefault}
-        className={classnames(styles.spinner, { [styles.delay]: delay })}
+        aria-label={accessibilityLabel}
+        className={classnames(styles.spinner, { [styles.exit]: !show })}
+        onAnimationEnd={onExitAnimationEnd}
         style={
           {
+            '--g-enter-delay': delay ? '300ms' : undefined,
             '--g-size': `${SIZE_NAME_TO_PIXEL[size]}px`,
           } as React.CSSProperties
         }
       >
-        <div className={styles.spinnerFrame}>
-          <div className={styles.dot1} />
-          <div className={styles.dot2} />
-          <div className={styles.dot3} />
-        </div>
+        <svg fill="none" viewBox="0 0 56 56" xmlns="http://www.w3.org/2000/svg">
+          <g style={{ '--g-start-x': '14px', '--g-start-y': '-8px' } as React.CSSProperties}>
+            <circle cx="14" cy="36" r="8" />
+          </g>
+          <g style={{ '--g-start-x': '0px', '--g-start-y': '16px' } as React.CSSProperties}>
+            <circle cx="28" cy="12" r="8" />
+          </g>
+          <g style={{ '--g-start-x': '-14px', '--g-start-y': '-8px' } as React.CSSProperties}>
+            <circle cx="42" cy="36" r="8" />
+          </g>
+        </svg>
       </div>
     </Box>
   );
 }
 
+type Props = {
+  accessibilityLabel?: string;
+  delay?: boolean;
+  show: boolean;
+  size?: 'sm' | 'md' | 'lg';
+};
+
+export default function Spinner({
+  accessibilityLabel,
+  delay = true,
+  show: showProp,
+  size = 'md',
+}: Props) {
+  const [show, setShow] = useState(showProp);
```

Comment:
i dont think this is the best API

1. the behavior is opaque, it's tied to the show prop
2. however, engineers dont use the show prop properly... ![Screenshot by Dropbox Capture](https://github.com/user-attachments/assets/cbc1d361-16ea-4d56-9088-1d9c1ad71179) 
3. the animations is not controlled and it can't be controlled externally or affect external state

Reply:
In terms of the API, I don't think anything has changed. This change doesn't effect the existing instances regardless of the choice of use:

1. If developers are conditionally rendering it, they can keep doing so. Spinner will just not show the outro animation.
2. If developers do want to have outro animation, they will need to control it by `show` prop.
3. If developers want to control it by `show` prop but don't want the outro animation:
  a. The can change the usage to conditional rendering.
  b. We can add another prop to disable outro animation.

Reply:
**decided to address it later